std/linked_list.zig

```diff
@@ -0,0 +1,271 @@
+const debug = @import("debug.zig");
+const assert = debug.assert;
+const mem = @import("mem.zig");
+const Allocator = mem.Allocator;
+
+/// Generic doubly linked list.
+pub fn LinkedList(comptime T: type) -> type {
+    struct {
+        const List = this;
+
+        /// Node inside the linked list wrapping the actual data.
+        const Node = struct {
```

Comment:
Probably. Using the list requires knowledge of the internals of `Node` anyway (the `data` field)